### PR TITLE
Parse Pinal county public health for AZ

### LINKED ISSUE #281: parse az/pinal_ph_vaccinelocations_gov
[learn our pipeline: parse](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#parse)

Blocked by #279

Parse existing data into [`.ndjson`](http://ndjson.org/).

Read all files in the directory passed as the second argument (`sys.argv[2]`), convert them to `.ndjson`, and then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the parse stage and how to get set up for development!

### Tips

1. Fetch data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch az/pinal_ph_vaccinelocations_gov
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest parse az/pinal_ph_vaccinelocations_gov
    ```

### Example
[Fetched files for `md/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/raw) are converted to [parsed files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/md/arcgis/parsed)

### PR DESCRIPTION
**Pull request by cdjj**
| Key Details |
|-|
| Resolves #281  |
State: AZ |
Site:  |

## Data sample
```json
{"providerName": "San Tan Valley Pinal County Public Health Clinic", "city": "San Tan Valley "}
{"providerName": "Mammoth Pinal County Public Health Clinic", "city": "Mammoth"}
{"providerName": "Casa Grande Pinal County Public Health Clinic", "city": "Casa Grande"}
{"providerName": "Apache Junction Pinal County Public Health Clinic", "city": "Apache Junction"}
{"providerName": "Coolidge Pinal County Public Health Clinic", "city": "Coolidge"}
{"providerName": "Maricopa Pinal County Public Health Clinic", "city": "Maricopa"}
{"providerName": "Embry Health TestNOW (and Vaccines)", "city": "San Tan Valley", "address": "32375 N Gantzel RD, San Tan Valley, AZ 85143", "website": "https://app.backstagemedical.co/widgets/#/schedule-appointment?pId=9567&widgetType=allopen&typeId=1547&count=300", "phoneNumber": "(480) 376-2170"}
```

## Before Opening a PR
- [X] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Review by eliblock — CHANGES_REQUESTED** [review]
This parser looks like a great start, but I think we're missing several of the sites [listed on the fetch'd page](https://www.pinalcountyaz.gov/publichealth/CoronaVirus/Pages/vaccinelocations.aspx). Specifically, it looks like there's about 30 parsed rows right now, but I count over 50 sites on the original page.

Could you take another look at why some of the sites might be getting dropped?

File: vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/parse.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+MODERNA_REGEX = re.compile(r".*Moderna.*")
+PFIZER_REGEX = re.compile(r".*Pfizer.*")
+
+
+def extractAddress(s):
+    candidates = s.select('a[href^="https://www.google.com/maps/"]')
+    for c in candidates:
+        address = c.string.strip()
+        address = address.replace("\n", "")
+        if address.startswith("Dexcom"):  # Special case, they don't give the address.
```
[the remark this answers is not in the document]
Comment (cdjj) [reply vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/parse.py:23]: There won't be extra --- this is a company warehouse that's being used as an indoor drive-through site b/c of heat outside, and they've only got the one warehouse (https://azgovernor.gov/governor/news/2021/03/indoor-drive-thru-covid-19-vaccination-site-mesa-will-maintain-vital-east).  There will be additional indoor drive-throughs at other sites, but they'll have other names.

**Comment by cdjj** [comment]
> Could you take another look at why some of the sites might be getting dropped?

It looks like they edit this by hand and have a stray `</tr>` in there, which confuses Beautiful Soup into ignoring anything after it in the table.  Taking a look to see if there's any way to tell it to accept invalid HTML....

```
  <td>
  <a href="https://www.mhealthappointments.com/covidappt" target="_blank">Register Here</a> <br> <br>
  <b>Administering Janssen-limited supply</b>
  </td>
  <td></td>
  </tr> 
   
  </tr>
   
  <tr>
  <td style="background-color:#FFFF00;">Federal</td>
  <td> Safeway/Albertsons </td>
```

**Comment by cdjj** [comment]
OK, swapping the parser looks like it ignores the spurious `</tr>` and finds 52 sites.

File: vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/parse.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cdjj) [reply vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/parse.py:23]: (Also, the text we get is "Dexcom Site", so there's not much more to disambiguate with....)

[the remark this answers is not in the document]
Comment (eliblock) [reply vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/parse.py:23]: Thank you for doing the extra research!

**Review by eliblock — APPROVED** [review]
Thank you for taking a second look!

**Pull request merged**